clang/lib/Sema/SemaChecking.cpp

```diff
@@ -4461,6 +4462,23 @@ bool Sema::CheckRISCVBuiltinFunctionCall(const TargetInfo &TI,
   return false;
 }
 
+bool Sema::CheckRISCVCOREVBuiltinFunctionCall(unsigned BuiltinID,
+                                           CallExpr *TheCall) {
+  // For intrinsics which take an immediate value as part of the instruction,
+  // range check them here.
+  switch (BuiltinID) {
+  default: return false;
+  case RISCVCOREV::BI__builtin_riscv_cv_simd_extract_h:
+  case RISCVCOREV::BI__builtin_riscv_cv_simd_extract_b:
+  case RISCVCOREV::BI__builtin_riscv_cv_simd_extractu_h:
+  case RISCVCOREV::BI__builtin_riscv_cv_simd_extractu_b:
+    return SemaBuiltinConstantArgRange(TheCall, 1, 0, 63);
```

Comment:
I don't think we should use 63 as range limit. These instructions are supposed to be used with values [0,1,2,3] for *_b and [0,1] *_h to select the bytes or half words. I'd make that clear straight away in the front-end, otherwise we may need to truncate or crash or silently mis-select the byte/half word later. So I'd say:
for extract_h and extractu_h:
```
  return SemaBuiltinConstantArgRange(TheCall, 1, 0, 1);
```
while for extract_b and extractu_b:
```
  return SemaBuiltinConstantArgRange(TheCall, 1, 0, 3);
```

Reply:
Thanks for the feedback. I'll make the changes accordingly.

Reply:
I have fixed the immediate operand ranges. I'll edit the tests now.

Reply:
I think I clang-formatted the file too. Is it fine? Or should I revert the changes?